File: 242. 有效的字母异位词.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        int dp[26]={0};
        
        for(int i=0;i<s.size();i++){
            dp[s[i]-'a']++;
        }
        
        for(int i=0;i<t.size();i++){
            dp[t[i]-'a']--;
        }
        
        for(int i=0;i<26;i++){
            if(dp[i]!=0)
                return false;
        }
        return true;
    }
};

File: js/242. 有效的字母异位词.js
var isAnagram = function(s, t) {
    if(s.length!==t.length)
        return false
    let hash={}
    for(let i=0;i<s.length;i++){
        hash[s[i]]=(hash[s[i]]|0)+1
    }
    for(let i=0;i<t.length;i++){
        if(hash[t[i]]===undefined)
            return false
        hash[t[i]]--
        // 检测 t 是否有多余 or 不同字符
        hash[t[i]]===0&&(hash[t[i]]=undefined)
    }
    return true
};
